File: webserv/cgi-bin/greet.py
#!/usr/bin/env python

import cgi

print("Content-Type: text/html")  # HTML is following
print()                           # Blank line, end of headers

# Create instance of FieldStorage
form = cgi.FieldStorage()

# Get data from fields
name = form.getvalue('name', 'Unknown')
age = form.getvalue('age', 'Unknown')

# Generate the HTML content
print(f"<html><head><title>CGI Script</title></head>")
print(f"<body>")
print(f"<h1>CGI Script Response</h1>")
print(f"<p>Name: {name}</p>")
print(f"<p>Age: {age}</p>")
print(f"</body></html>")

File: webserv/cgi-bin/greet.php
#!/usr/bin/env php
<?php
// Ensure that the content type header is set to HTML
header('Content-Type: text/html');

// Parse the QUERY_STRING environment variable
parse_str(getenv('QUERY_STRING'), $params);

// Retrieve data from the query parameters
$name = isset($params['name']) ? $params['name'] : 'Unknown';
$age = isset($params['age']) ? $params['age'] : 'Unknown';

// Generate the HTML content
echo "<html><head><title>CGI Script</title></head>\n";
echo "<body>\n";
echo "<h1>CGI Script Response</h1>\n";
echo "<p>Name: " . htmlspecialchars($name) . "</p>\n";
echo "<p>Age: " . htmlspecialchars($age) . "</p>\n";
echo "</body></html>";
?>
